src/storage/v2/storage.cpp

```diff
@@ -2014,16 +2030,19 @@ void Storage::RestoreReplicas() {
     MG_ASSERT(replica_status.name == replica_name, "Expected replica name is '{}', but got '{}'", replica_status.name,
               replica_name);
 
-    auto ret = RegisterReplica(std::move(replica_status.name),
-                               {std::move(replica_status.ip_address), replica_status.port}, replica_status.sync_mode,
-                               {
-                                   .replica_check_frequency = replica_status.replica_check_frequency,
-                                   .ssl = replica_status.ssl,
-                               });
+    auto ret =
+        RegisterReplica(std::move(replica_status.name), {std::move(replica_status.ip_address), replica_status.port},
+                        replica_status.sync_mode, storage::replication::RegistrationMode::RECOVERY_AT_STARTUP,
+                        {
+                            .replica_check_frequency = replica_status.replica_check_frequency,
+                            .ssl = replica_status.ssl,
+                        });
     if (ret.HasError()) {
-      LOG_FATAL("Failure when restoring replica {}: {}.", replica_name, ret.GetError());
+      spdlog::error("Failure when restoring replica {}: {}.", replica_name,
+                    RegisterReplicaErrorToString(ret.GetError()));
+    } else {
+      spdlog::info("Replica {} restored.", replica_name);
```

Comment:
As we just discussed this should still be a `LOG_FATAL`. The above `RegisterReplica` call won't fail because of connection issues, thus every possible error means our knowledge of the replicas is corrupted, we shouldn't continue our startup.

Reply:
Done